ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/AdvertisingTests.kt

```diff
@@ -0,0 +1,133 @@
+        val player = mainHandler.postWaiting {
+            val player = Player(
+                    context,
+                    PlayerConfig(
+                            key = BITMOVIN_PLAYER_LICENSE_KEY,
+                            advertisingConfig = AdvertisingConfig(
+                                    AdItem(AdSource(AdSourceType.Ima, VMAP_PREROLL_SINGLE_TAG)),
+                            ),
+                            playbackConfig = PlaybackConfig(
+                                    isAutoplayEnabled = true,
+                            ),
+                    ),
+                    analyticsConfig = AnalyticsPlayerConfig.Disabled,
+            )
+            val convivaAnalyticsIntegration = ConvivaAnalyticsIntegration(
+                    player,
+                    CONVIVA_CUSTOMER_KEY,
+                    context,
+                    ConvivaConfig().apply {
+                        isDebugLoggingEnabled = true
+                        gatewayUrl = CONVIVA_GATEWAY_URL
+                    },
+            )
+
+            convivaAnalyticsIntegration.updateContentMetadata(
+                    MetadataOverrides()
+                            .apply {
+                                applicationName = "Bitmovin Android Conviva integration example app"
+                                viewerId = "testViewerId"
+                            }
+            )
+            player
+        }
+
+        mainHandler.postWaiting { player.load(ART_OF_MOTION_DASH) }
+        player.expectEvent<PlayerEvent.TimeChanged> { it.time > 5.0 }
+
+        // pause player for a second and resume playback
+        mainHandler.postWaiting { player.pause() }
+        runBlocking { delay(1000) }
+        mainHandler.postWaiting { player.play() }
+
+        // wait for the ad break to finish and play main content for five more seconds
+        player.expectEvent<PlayerEvent.AdBreakFinished>()
+        player.expectEvent<PlayerEvent.TimeChanged> { it.time > 5.0 }
+
+        mainHandler.postWaiting { player.destroy() }
+        runBlocking { delay(1000) }
+    }
+}
+
+/**
+ * Subscribes to an [Event] on the [Player] and suspends until the event is emitted.
+ * Optionally a [condition] can be provided to filter the emitted events.
+ */
+private inline fun <reified T : Event> EventEmitter<Event>.expectEvent(
+        crossinline condition: (T) -> Boolean = { true }
+) = runBlocking {
+    suspendCoroutine { continuation ->
+        lateinit var action: ((T) -> Unit)
+        action = {
+            if (condition(it)) {
+                off(action)
+                continuation.resume(Unit)
+            }
+        }
+        on<T>(action)
+    }
+}
+
+/**
+ * Posts a [block] of code to the main thread and suspends until it is executed.
+ */
+private inline fun <T> Handler.postWaiting(crossinline block: () -> T) = runBlocking {
+    suspendCoroutine { continuation ->
+        post { continuation.resume(block()) }
+    }
+}
```

Comment:
The test helpers are very limited at the moment, hence I kept them in this test file. Should we decide to add more tests we can make them more generic and extract them to a better location.